google/cloud/sql/connector/instance.py

```diff
@@ -231,6 +233,10 @@ async def _perform_refresh(self) -> ConnectionInfo:
 
             logger.debug(f"['{self._instance_connection_string}']: Creating context")
 
+            # before making Cloud SQL Admin API calls, refresh creds
+            if not self._client._credentials.token_state == TokenState.FRESH:
+                self._client._credentials.refresh(requests.Request())
+
```

Comment:
moved checking of credentials up a level into the refresh operation and out of the individual API calls, this way we are checking once instead of twice.